src/AIModule.cpp

```diff
@@ -1,95 +1,132 @@
 #include <AIModule.h>
 #include "Cast.hpp"
 
-class AIModuleWrapper : public BWAPI::AIModule
-{
+template<typename T>
+void checkPointer(T* pointer) {
+    bool isValid = true;
+    if (pointer == nullptr) {
+        isValid = true;
+    }
+    // Check whether high 16 bits are correctly sign-extended
+    auto p = (uintptr_t) pointer;
+    int highestUsedBit = (p >> 47) & 1;
```

Comment:
no need to check it if pointer == nullptr

Reply:
NULL address is correctly sign-extended